cli/cmd/generate.go

```diff
@@ -388,6 +391,60 @@ func addWorkloadOwnerKeyToManifest(manifst *manifest.Manifest, keyPath string) e
 	return nil
 }
 
+type logTranslator struct {
+	r         *io.PipeReader
+	w         *io.PipeWriter
+	logger    *slog.Logger
+	stopDoneC chan struct{}
+}
+
+func newLogTranslator(logger *slog.Logger) logTranslator {
+	r, w := io.Pipe()
+	l := logTranslator{
+		r:         r,
+		w:         w,
+		logger:    logger,
+		stopDoneC: make(chan struct{}),
+	}
+	l.startTranslate()
+	return l
+}
+
+func (l logTranslator) Write(p []byte) (n int, err error) {
+	return l.w.Write(p)
+}
+
+var genpolicyLogPrefixReg = regexp.MustCompile(`^\[[^\]\s]+\s+(\w+)\s+([^\]\s]+)\] (.*)`)
+
+func (l logTranslator) startTranslate() {
+	go func() {
+		defer close(l.stopDoneC)
+		scanner := bufio.NewScanner(l.r)
+		for scanner.Scan() {
+			line := scanner.Text()
+			match := genpolicyLogPrefixReg.FindStringSubmatch(line)
+			if len(match) != 4 {
+				// genpolicy prints some warnings without the logger
+				l.logger.Warn(line)
+			} else {
+				switch match[1] {
+				case "ERROR":
+					l.logger.Error(match[3], "position", match[2])
+				case "WARN":
+					l.logger.Warn(match[3], "position", match[2])
+				case "INFO": // prints quite a lot, only show on debug
+					l.logger.Debug(match[3], "position", match[2])
+				}
```

Comment:
How is this different from setting `RUST_LOG` based on the configured Contrast log level and then logging all of stderr?

Reply:
sgtm